From: Tony Camuso <tcamuso@redhat.com>
Date: Thu, 28 Aug 2008 22:31:55 -0400
Subject: [net] NetXen: remove performance optimization fix
Message-id: 48B75F9B.7050204@redhat.com
O-Subject: [RHEL5.3 PATCH]BZ 457958 NetXen Critical Bug Fix
Bugzilla: 457958
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

Bugzilla
========
https://bugzilla.redhat.com/show_bug.cgi?id=457958

Description
===========

Critical Bug Fix

On AMD servers with certain versions of the Serverworks HT2100, when the
upstream driver loads, the PCIe bus may hang. This patch removes a
performance optimization which causes the hang.

No kabi symbols affected.

Test Status
===========
NetXen has successfully conducted the following tests with these
patches applied.
. Data Integrity Tests
. Network Functional Tests
. Long run Tests
. Interoperability Tests
. Performance Tests
. Regression test
. Integration test
. Unit test

Stat
====
  drivers/net/netxen/netxen_nic_main.c |   10 ----------
  1 files changed, 0 insertions(+), 10 deletions(-)

Signed-off-by: Dhananjay Phadke <dhananjay@netxen.com>
Signed-off-by: Tony Camuso <tcamuso@redhat.com>

Upstream Status
===============
The patch has been submitted upstream, but was only posted a day ago. There
has been no action yet, but we fully expect this patch to be accepted
upstream.

Without this patch, AMD systems with certain versions of the HT2100 chipset
could hang.

Please ACK

diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index de6f40b..86a141f 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -366,16 +366,6 @@ static void netxen_pcie_strap_init(struct netxen_adapter *adapter)
 	int i, pos;
 	struct pci_dev *pdev;
 
-	pdev = pci_get_device(0x1166, 0x0140, NULL);
-	if (pdev) {
-		pci_dev_put(pdev);
-		adapter->hw_read_wx(adapter,
-			NETXEN_PCIE_REG(PCIE_TGT_SPLIT_CHICKEN), &chicken, 4);
-		chicken |= 0x4000;
-		adapter->hw_write_wx(adapter,
-			NETXEN_PCIE_REG(PCIE_TGT_SPLIT_CHICKEN), &chicken, 4);
-	}
-
 	pdev = adapter->pdev;
 
 	adapter->hw_read_wx(adapter,
